ngraph/core/include/ngraph/op/constant.hpp

```diff
@@ -352,13 +404,112 @@ namespace ngraph
+                    const auto source_end = std::next(source_begin, (element_number + 1) / 2);
+                    const auto round_element_no =
+                        element_number % 2 ? element_number + 1 : element_number;
+                    output.reserve(round_element_no); // adds 1 more elements here?
+                    std::for_each(source_begin, source_end, [&](IN_T c) {
+                        for (const auto i : {4, 0})
+                        {
+                            const uint8_t i4data = (c >> i) & 0x0F;
+                            const bool is_negative_number = (i4data >> 3) & 0b1;
+                            const int8_t data = is_negative_number ? i4data | 0xF0 : i4data;
+                            output.push_back(data);
+                        }
+                    });
+                    output.resize(element_number);
+                }
+
                 template <element::Type_t Type,
                           typename T,
-                          typename DataStorageType = fundamental_type_for<Type>>
+                          typename DataStorageType = fundamental_type_for<Type>,
+                          typename std::enable_if<Type != element::Type_t::u1 &&
+                                                      Type != element::Type_t::u4 &&
+                                                      Type != element::Type_t::i4,
+                                                  bool>::type = true>
                 void fill_data(const T& value)
                 {
                     const auto size = shape_size(m_shape);
-                    std::fill_n(get_data_ptr_nc<Type>(), size, static_cast<DataStorageType>(value));
+                    const auto v = static_cast<DataStorageType>(value);
+                    std::fill_n(get_data_ptr_nc<Type>(), size, v);
+                }
+
+                template <element::Type_t Type,
+                          typename T,
+                          typename DataStorageType = fundamental_type_for<Type>,
+                          typename std::enable_if<Type == element::Type_t::u1, bool>::type = true>
+                void fill_data(const T& value)
+                {
+                    const auto size = (shape_size(m_shape) + 7) / 8;
+                    const DataStorageType v = value ? 0xFF : 0x00;
+                    std::fill_n(get_data_ptr_nc<Type>(), size, v);
+                }
+
+                template <element::Type_t Type,
+                          typename T,
+                          typename DataStorageType = fundamental_type_for<Type>,
+                          typename std::enable_if<Type == element::Type_t::u4 ||
+                                                      Type == element::Type_t::i4,
+                                                  bool>::type = true>
+                void fill_data(const T& value)
+                {
+                    const auto size = (shape_size(m_shape) + 1) / 2;
```

Comment:
I think we need to throw an exception if original value is out of range for LP types.

Reply:
OK